### ISSUE TITLE
PyMultinest broken on macOS Big Sur

### ISSUE THREAD

**Issue by vetlewi** [oslocyclotronlab/ompy#169]
The PyMultinest will cause segfaults whenever running on macOS Big Sur. See issue in PyMultinest repo: https://github.com/JohannesBuchner/PyMultiNest/issues/174

**Comment by fzeiser** [comment]
Comment: As PyMultinest is used in ompy in the normalization, this will lead to an error when using the normalization classes of ompy downstream. 

Thank you for reporting this.

**Comment by JohannesBuchner** [comment]
You can try UltraNest instead, which should be easier to install. https://johannesbuchner.github.io/UltraNest/

**Comment by fzeiser** [comment]
@JohannesBuchner: Thanks a lot for the suggestion, we may try it out. In the meantime, this might be an alternative for mac users.

**Comment by fzeiser** [comment]
@vetlewi : did you try the example with the garbage collection (see latest comments in the `PyMultiNest` issue)?

**Comment by vetlewi** [comment]
> @vetlewi : did you try the example with the garbage collection (see latest comments in the `PyMultiNest` issue)?

Well, when I ran the PyMultinest examples with a debugger (in my case I use PyCharm, not sure what it exactly uses under the hood) it doesn't give seg fault, but running without the debugger it does. Makes debugging super easy!

**Comment by vetlewi** [comment]
> > @vetlewi : did you try the example with the garbage collection (see latest comments in the `PyMultiNest` issue)?
> 
> Well, when I ran the PyMultinest examples with a debugger (in my case I use PyCharm, not sure what it exactly uses under the hood) it doesn't give seg fault, but running without the debugger it does. Makes debugging super easy!

Just to expand a little; running in debugging mode is in no way a solution to the problem!

**Comment by fzeiser** [comment]
I totally agree. How urgent do you think that this is? If we can wait some time, it would be nice to see if the problem "disappears" soon, instead of changing the library *for everyone*.

**Comment by fzeiser** [comment]
Which python version do you use on Big Sur? Do you think that using another version might help? (It must be >=3.7)

**Comment by vetlewi** [comment]
> Which python version do you use on Big Sur? Do you think that using another version might help? (It must be >=3.7)

Tried 3.7, 3.8, 3.9.

**Comment by grburgess** [comment]
I've also had problems with this. It seems to be an issue in how the ctypes bindings are handled in Big Sur. They have gotten stricter  and I think it can be solved by changing a few things.

**Comment by vetlewi** [comment]
I haven't updated anything, but for some reason it now decided to work... This is so strange! I'm so confused right now...
